### PR TITLE
enable cloud provider to scale cluster

### PR DESCRIPTION
**Pull request by xiaohan1202**

### CODE REVIEW

File: controllers/cluster/controllers/cluster_controller.go [hunk 1]
```diff
@@ -18,9 +18,16 @@ package controllers
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"path/filepath"
 	"time"
 
+	"github.com/labring/sealos/pkg/utils/logger"
```
Comment (zzjin) [inline controllers/cluster/controllers/cluster_controller.go:26]: Controllers can direct use kubebuilder‘s reconcile logger to log informations, there is no such need to use sealos's `pkg/logger`, witch will cause package all `sealos/pkg` code, seems too large for controller.

**Review by zzjin — CHANGES_REQUESTED** [review]
Need remove requirements.

**Review by fanux — APPROVED** [review]

File: controllers/cluster/controllers/cluster_controller.go [hunk 2]
```diff
@@ -18,9 +18,16 @@ package controllers
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"path/filepath"
 	"time"
 
+	"github.com/go-logr/logr"
```
Comment (zzjin) [inline controllers/cluster/controllers/cluster_controller.go:26]: image
Use buildin log like user controller.

File: go.work [hunk 3]
```diff
@@ -16,3 +16,11 @@ use (
...
 	./webhooks/whitelist
 )
+
+replace k8s.io/apimachinery => k8s.io/apimachinery v0.24.3
+
+replace k8s.io/code-generator => k8s.io/code-generator v0.24.3
+
+replace k8s.io/cri-api => k8s.io/cri-api v0.24.3
+
+replace k8s.io/mount-utils => k8s.io/mount-utils v0.24.3
```
Comment (zzjin) [inline go.work:26]: These changes can also be removed?

File: controllers/cluster/controllers/cluster_controller.go [hunk 4]
```diff
@@ -174,22 +245,27 @@ func getSealosVersion(cluster *v1.Cluster) string {
...
+			return fmt.Errorf("wait node %v ssh ready failed: %v", ip, err)
+		}
+	}
+	return nil
+}
+
+func getSSHclient(infra *infrav1.Infra) ssh.Interface {
 	s := &v1beta1.SSH{
-		User:   infra.Spec.SSH.User,
+		User:   "root",
```
Comment (zzjin) [inline controllers/cluster/controllers/cluster_controller.go:259]: Remove hardcoded `root` with `infra.Spec.SSH.User` or `defaultUser`?

**Review by zzjin — APPROVED** [review]

**Pull request merged**